compiler/damlc/lib/DA/Cli/Damlc/Packaging.hs

```diff
@@ -80,9 +80,9 @@ createProjectPackageDb projectRoot (disableScenarioService -> opts) thisSdkVer m
     deps <- expandSdkPackages (optDamlLfVersion opts) (filter (`notElem` basePackages) deps)
     depsExtracted <- mapM extractDar deps
 
-    let uniqSdkVersions = nubSort $ unPackageSdkVersion thisSdkVer : map edSdkVersions depsExtracted
+    let uniqSdkVersions = nubSort $ thisSdkVer : map edSdkVersions depsExtracted
```

Comment:
I’m confused by this. `thisSdkVer` is now in the GHC format. However `edSdkVersions` is the one read from the manifest which with this change should now be the actual SDK version. What am I missing?

Reply:
You're missing nothing. I think I'm missing something now though. Where did we actually need the "sanitized" version? To me this looks like the only place we use the `PackageSdkVersion` is in the manifest.

Reply:
I’m a bit confused why CI even passes. Shouldn’t the comparison that dependencies have the same sdk version fail with this change? Does it only work because this is not a release build so the SDK version is 0.0.0? What happens if you set `DAML_SDK_RELEASE_VERSION` to a snapshot?

As to where this is used: We need to use the mangled version when building `daml-stdlib` and `daml-prim` but that is set in a Bazel build rather than our Haskell code. I’m not quite sure if we need the mangled version in our Haskell code.

Reply:
It looks like we don't need the mangled version in our Haskell code at all. I've removed it and ran
```sh
DAML_SDK_RELEASE_VERSION=0.0.0-snapshot.20200930.5312.0.b9a1905d daml-sdk-head --skip-jars
```
Then I've built two projects `foo` and `bar` using that SDK where `bar` depends on `foo` and everything worked fine.

I'm wondering if we should add a test for this but I assume it doesn't make too much sense since the release PRs for snapshot releases would catch such issues.

Reply:
Ah, and
```sh
DAML_SDK_RELEASE_VERSION=0.0.0-snapshot.20200930.5312.0.b9a1905d bt //compiler/damlc/tests:packaging
```
works as well.